pkg/executor/aggfuncs/func_group_concat.go

```diff
@@ -429,6 +429,8 @@ type partialResult4GroupConcatOrder struct {
 
 type groupConcatOrder struct {
 	baseGroupConcat4String
+	ctors []collate.Collator
```

Comment:
Why add these 2 new fields?

Reply:
Because in `func (e *groupConcatOrder) AllocPartialResult()`, it'll use these fields to allocate partial result. Previously, it was generated in `AllocPartialResult`. However, add an argument `EvalContext` to `AllocPartialResult` will lead to tons of changes, so I choose to calculate the `ctors` and `desc` when this struct is constructed.